core/dnsserver/register.go

```diff
@@ -78,18 +82,14 @@ func (h *dnsContext) InspectServerBlocks(sourceFile string, serverBlocks []caddy
 			}
 
 			ones, bits := za.IPNet.Mask.Size()
-			if (bits-ones)%8 != 0 { // only do this for non-octet boundaries
-				cfg.FilterFunc = func(s string) bool {
```

Comment:
I removed this closure because the variable `za` was being set to the last `zoneAddr` seen in `InspectServerBlocks` due to the lexical scoping of variables within a closure.